### ISSUE TITLE
[Bug] LinearAttention padding keys affect normalization and gradients

### ISSUE THREAD

**Issue by Lexwzp0** [fla-org/flash-linear-attention#1263]
### Checklist

- [x] I have checked the FAQs and existing issues for similar problems.
- [x] This report is in English.

### Describe the Bug

With `do_feature_map_norm=True`, `LinearAttention` masks V but leaves padding keys in the cumulative normalization denominator. Adding left padding changes valid outputs and input gradients; masked inputs also receive nonzero gradients.

This reproduces with the upstream layer alone at `864a87f6ce5be8828bef81eb22baafd41937cdf2`, using synthetic inputs:

| Metric | FP32 | BF16 |
| --- | ---: | ---: |
| Valid output relative RMS error | 0.734127 | 0.734251 |
| Valid input gradient relative RMS error | 0.741432 | 0.741352 |
| Nonzero masked input gradient entries | 4096 | 4096 |

### Steps to Reproduce

```python
"""Check the padding invariant of normalized linear attention without a model or position encoding."""

import json

import torch

from fla.layers.linear_attn import LinearAttention


def relative_rms(reference, actual):
    reference, actual = reference.float(), actual.float()
    return ((reference - actual).square().mean() / reference.square().mean()).sqrt().item()


results = []
for dtype in (torch.float32, torch.bfloat16):
    torch.manual_seed(42)
    layer = LinearAttention(
        hidden_size=128, num_heads=2, feature_map='elu', do_feature_map_norm=True,
        output_norm='identity', mode='chunk', layer_idx=0,
    ).to(device='cuda', dtype=dtype)
    x = torch.randn(1, 129, 128, device='cuda', dtype=dtype)
    padding = torch.randn(1, 32, 128, device='cuda', dtype=dtype)
    padded = torch.cat((padding, x), dim=1)
    mask = torch.ones(1, 161, device='cuda', dtype=torch.long)
    mask[:, :32] = 0
    with torch.no_grad():
        reference = layer(x)[0]
        actual = layer(padded, attention_mask=mask)[0][:, 32:]
    x = x.requires_grad_()
    padded = padded.requires_grad_()
    reference_train = layer(x)[0]
    actual_train = layer(padded, attention_mask=mask)[0][:, 32:]
    do = torch.randn_like(reference_train)
    reference_dx, = torch.autograd.grad((reference_train * do).sum(), x)
    actual_dx, = torch.autograd.grad((actual_train * do).sum(), padded)
    results.append({
        'dtype': str(dtype),
        'output_relative_rms': relative_rms(reference, actual),
        'training_output_relative_rms': relative_rms(reference_train, actual_train),
        'input_gradient_relative_rms': relative_rms(reference_dx, actual_dx[:, 32:]),
        'masked_input_gradient_nonzero': int(actual_dx[:, :32].count_nonzero()),
    })
print(json.dumps(results, indent=2))
```

### Expected Behavior

Adding masked padding should preserve valid outputs and gradients, and masked inputs should receive zero gradient.

### Environment

NVIDIA RTX PRO 6000 Blackwell Server Edition; PyTorch 2.9.0+cu128; Triton 3.5.0. Tested with FLA's default Triton implementation.

### Possible Fix

Reuse `unpad_hidden_states` / `repad_hidden_states` and pass the resulting `cu_seqlens` to the existing operators, as other recurrent layers do. Preserve the original query length for cache offsets.

<details>
<summary>Candidate patch</summary>

```diff
--- a/fla/layers/linear_attn.py
+++ b/fla/layers/linear_attn.py
@@ -15,7 +15,7 @@
 import torch.nn.functional as F
 from einops import rearrange, repeat
 
-from fla.layers.utils import get_layer_cache, update_layer_cache
+from fla.layers.utils import get_layer_cache, repad_hidden_states, unpad_hidden_states, update_layer_cache
 from fla.modules import RMSNorm
 from fla.modules.feature_map import DPFPFeatureMap, HadamardFeatureMap, HedgehogFeatureMap, T2RFeatureMap
 from fla.ops.linear_attn import chunk_linear_attn, fused_chunk_linear_attn, fused_recurrent_linear_attn
@@ -136,6 +136,7 @@
         output_attentions: bool | None = False,
         **kwargs,
     ) -> tuple[torch.Tensor, torch.Tensor | None, Cache | None]:
+        batch_size, q_len, _ = hidden_states.shape
         # Match other recurrent layers: use the recurrent kernel for decode/small chunks.
         if torch.is_grad_enabled():
             mode = 'chunk'
@@ -144,13 +145,12 @@
         else:
             mode = self.mode
         last_state = get_layer_cache(self, past_key_values)
+        cu_seqlens = kwargs.get('cu_seqlens')
+        hidden_states, indices, cu_seqlens = unpad_hidden_states(hidden_states, cu_seqlens, attention_mask, q_len)
 
         q = self.q_proj(hidden_states)
         k = self.k_proj(hidden_states)
         v = self.v_proj(hidden_states)
-
-        if attention_mask is not None:
-            v = v.mul(attention_mask[:, -v.shape[-2]:, None])
 
         q = rearrange(q, '... (h d) -> ... h d', d=self.head_k_dim)
         if self.num_kv_groups > 1:
@@ -177,6 +177,7 @@
                 initial_state=recurrent_state,
                 output_final_state=use_cache,
                 normalize=self.do_feature_map_norm,
+                cu_seqlens=cu_seqlens,
             )
         elif mode == 'fused_chunk':
             o, final_state = fused_chunk_linear_attn(
@@ -186,6 +187,7 @@
                 initial_state=recurrent_state,
                 output_final_state=use_cache,
                 normalize=self.do_feature_map_norm,
+                cu_seqlens=cu_seqlens,
             )
         elif mode == 'fused_recurrent':
             o, final_state = fused_recurrent_linear_attn(
@@ -195,6 +197,7 @@
                 initial_state=recurrent_state,
                 output_final_state=use_cache,
                 normalize=self.do_feature_map_norm,
+                cu_seqlens=cu_seqlens,
             )
         else:
             raise NotImplementedError
@@ -202,9 +205,10 @@
             self,
             past_key_values,
             recurrent_state=final_state,
-            offset=q.shape[1],
+            offset=q_len,
         )
         o = self.norm(o)
         o = rearrange(o, '... h d -> ... (h d)')
         o = self.o_proj(o)
+        o = repad_hidden_states(o, indices, batch_size, q_len)
         return o, None, past_key_values
```

</details>

With this patch, all three metrics above are zero in both dtypes. Twelve regression cases passed, covering lengths 31/129, padding lengths 7/32/64, FP32/BF16, forward, input/parameter gradients and cached continuation, with a `1e-3` relative RMS bound. These checks cover nonempty sequences with chunk training and chunk/recurrent inference; fused-chunk inference, fully padded sequences and other feature maps have not been validated.

The patch changes only the layer's padding handling; kernel arithmetic and checkpoint structure are unchanged. Would this approach make sense? Thanks!
